orbit/cmd/desktop/desktop.go

```diff
@@ -83,78 +48,89 @@ func main() {
 		myDeviceItem.Disable()
 		transparencyItem := systray.AddMenuItem("Transparency", "")
 
+		var insecureSkipVerify bool
+		if os.Getenv("FLEET_DESKTOP_INSECURE") != "" {
+			insecureSkipVerify = true
+		}
+
+		// TODO: figure out the right rootCA to pass to the client
+		client, err := service.NewDeviceClient(basePath, deviceToken, insecureSkipVerify, "")
+
+		if err != nil {
+			log.Printf("unable to initialize request client: %s", err)
+			os.Exit(1)
+		}
+
 		// Perform API test call to enable the "My device" item as soon
 		// as the device auth token is registered by Fleet.
-		go func() {
-			ticker := time.NewTicker(5 * time.Second)
-			defer ticker.Stop()
+		deviceEnabledChan := func() <-chan bool {
+			done := make(chan bool)
 
-			tr := http.DefaultTransport.(*http.Transport)
-			if os.Getenv("FLEET_DESKTOP_INSECURE") != "" {
-				// #nosec
-				tr.TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
-			}
-			client := &http.Client{
-				Transport: tr,
-			}
+			go func() {
+				ticker := time.NewTicker(5 * time.Second)
+				defer ticker.Stop()
 
-			for {
-				resp, err := client.Get(devTestURL.String())
-				if err != nil {
-					// To ease troubleshooting we set the tooltip as the error.
-					myDeviceItem.SetTooltip(err.Error())
-					log.Printf("get device URL: %s", err)
-				} else {
-					resp.Body.Close()
-					if resp.StatusCode == http.StatusOK {
+				for {
+					_, err := client.ListDevicePolicies()
+
+					switch {
+					case err == nil:
 						myDeviceItem.SetTitle("My device")
 						myDeviceItem.Enable()
 						myDeviceItem.SetTooltip("")
+						done <- true
 						return
+					case errors.Is(err, service.ErrMissingLicense):
+						myDeviceItem.SetTitle("My device")
```

Comment:
Missing call to `myDeviceItem.Enable()` here? Or if it's just the same code as above just use an `if err == nil || errors.Is(err, service.ErrMissingLicense)`.

Reply:
over-communicating, for some reason I thought we needed to disable the button if you don't have a valid license. Will update

Reply:
Currently the only premium feature on Fleet Desktop is the "notifications and display of host issues/failed-policies".
- Notification: Such notification currently is just a "colored red/green ball" (we'll add proper notifications soon).
- Display: The display of a "pPolicies" tab in the page when visiting "My device".

So, free users should still be able to see "My device" menu item enabled and when visiting the page they won't get a Policies "tab".